indico/MaKaC/common/output.py

```diff
@@ -494,40 +494,30 @@ def _generate_contrib_location_and_time(self, contrib, out):
             out.closeTag('datafield')
 
     def _generate_contrib_people(self, contrib, out, subcontrib=None):
-        aList = list(set(contrib.primary_authors) | set(contrib.secondary_authors))
         sList = list(subcontrib.person_links) if subcontrib else list(contrib.speakers)
-        list_ = {}
-        auth = list(contrib.primary_authors)
-        if auth:
-            auth = auth[0]
-            list_[auth] = ['Primary Author']
-        for user in aList:
-            if user in list_:
-                if user != auth:
-                    list_[user].append('Author')
+        users = {}
+        primary_authors = list(contrib.primary_authors)
+        secondary_authors = list(contrib.secondary_authors)
+        for primary_author in primary_authors:
+            users[primary_author] = ['Primary Author']
+
+        for user in secondary_authors:
+            if user in users:
+                users[user].append('Author')
             else:
-                list_[user] = ['Author']
+                users[user] = ['Author']
 
         for user in sList:
-            if user in list_:
-                list_[user].append('Speaker')
+            if user in users:
+                users[user].append('Speaker')
             else:
-                list_[user] = ['Speaker']
+                users[user] = ['Speaker']
 
-        if auth:
-            user = auth
-            out.openTag('datafield', [['tag', '100'], ['ind1', ' '], ['ind2', ' ']])
-            out.writeTag('subfield', u'{} {}'.format(auth.last_name, auth.first_name), [['code', 'a']])
-            for val in list_[user]:
-                out.writeTag('subfield', val, [['code', 'e']])
-            out.writeTag('subfield', auth.affiliation, [['code', 'u']])
-            out.closeTag('datafield')
-            del list_[auth]
-
-        for user in list_.keys():
-            out.openTag('datafield', [['tag', '700'], ['ind1', ' '], ['ind2', ' ']])
+        for user in users.keys():
```

Comment:
you get the same by iterating over `users` without `.keys()` - also, I'd use iteritems instead of accessing `users[user]` later